File: short_problems/Python/utopian_tree.py
#!/usr/local/bin/python3

def height(n):
    if n == 0:
        return 1
    elif n & 1:
        return (1 << (((n + 1) >> 1) + 1)) - 2
    return (1 << ((n >> 1) + 1)) - 1

if __name__ == "__main__":
    num_cases = int(input())
    for i in range(num_cases):
        n = int(input())
        print(height(n))

File: short_problems/C++/utopian_tree.cpp
#include <iostream>
using namespace std;

int height(int n) {
    if (n == 0) {
        return 1;
    } else if (n & 1) {
        return (1 << (((n + 1) >> 1) + 1)) - 2;
    }
    return (1 << ((n >> 1) + 1)) - 1;
}

int main() {
    int T;
    cin >> T;
    while (T--) {
        int n;
        cin >> n;
        cout << height(n) << endl;
    }
}
